metron-platform/metron-enrichment/src/main/java/org/apache/metron/enrichment/bolt/GenericEnrichmentBolt.java

```diff
@@ -149,9 +154,10 @@ public JSONObject load(CacheKey key) throws Exception {
     cache = CacheBuilder.newBuilder().maximumSize(maxCacheSize)
             .expireAfterWrite(maxTimeRetain, TimeUnit.MINUTES)
             .build(loader);
+    GeoLiteDatabase.INSTANCE.update((String)getConfigurations().getGlobalConfig().get(GeoLiteDatabase.GEO_HDFS_FILE));
     boolean success = adapter.initializeAdapter();
```

Comment:
Does this mean that we will attempt to load the geo database into memory in every `GenericEnrichmentBolt`; even ones not doing geo-enrichment?  Maybe that is one reason, we have singleton for the GeoLiteDatabase; to avoid repeated initialization?

Along the same lines, If I choose to not do geo-enrichment, do I still need to have a valid Maxmind file in HDFS?  Or would that cause all GenericEnrichmentBolts to fail to initiailize?

Would it make sense to update the EnrichmentAdapter interface to accept configuration values, so that this initialization occurs in the GeoAdapter where it makes more sense?  Then only the bolts doing geo-enrichment attempt to initialize the geo data?